File: Exercism Codes/C#/SimpleCipher.cs
public class SimpleCipher
{
    public SimpleCipher()
    {
        Key = string.Join("", Enumerable.Range(0, 100).Select(x => (char)(new Random().Next(26) + 97)));
    }

    public SimpleCipher(string key) => Key = key;

    public string Key { get; set; }

    public string Encode(string plaintext) => string.Join("", plaintext.Select((x, idx) => (char)(97 + (((x - 97) + (Key[idx % Key.Length] - 97)) % 26))));

    public string Decode(string ciphertext)
    {
        return string.Join("", ciphertext.Select((x, idx) =>
        {
            int result = ((x - 97) - (Key[idx % Key.Length] - 97)) % 26;
            return (char)(97 + (result < 0 ? result + 26 : result));
        }));
    }
}

File: Exercism Codes/PHP/SimpleCipher.php
<?php

declare(strict_types=1);

class SimpleCipher
{
    public $key;

    public function __construct(string $key = null)
    {
        if ($key === null) {
            $this->key = implode("", array_map(fn($val): string =>  mb_chr(rand(97, 122)), range(0, 99)));
        }
        else {
            if ($key === "" || preg_match("/[A-Z0-9]/", $key)) {
                throw new InvalidArgumentException();
            }

            $this->key = $key;
        }
    }

    public function encode(string $plainText): string
    {
        $result = [];
        for ($i = 0; $i < strlen($plainText); $i++) {
            $char = (mb_ord($plainText[$i]) - 97) + (mb_ord($this->key[$i % strlen($this->key)]) - 97);
            $result[] = mb_chr(($char % 26) + 97);
        }
        return implode("", $result);
    }

    public function decode(string $cipherText): string
    {
        $result = [];
        for ($i = 0; $i < strlen($cipherText); $i++) {
            $char = (mb_ord($cipherText[$i]) - 97) - (mb_ord($this->key[$i % strlen($this->key)]) - 97);
            $result[] = mb_chr((($char % 26) + ($char < 0 ? 26 : 0)) + 97);
        }
        return implode("", $result);
    }
}